include/extractor/extractor_callbacks.hpp

```diff
@@ -84,6 +88,9 @@ class ExtractorCallbacks
 
     // destroys the internal laneDescriptionMap
     guidance::LaneDescriptionMap &&moveOutLaneDescriptionMap();
+
+    // destroys classes_map
+    ClassesMap &&moveOutClassMap();
```

Comment:
I don't understand the `R&& fn();` return value here. Shouldn't this be `R fn();` and internally we should probably set a `is_dead` boolean to allow only for calling it once?

The move-out here and with the lane map above is a bit ugly, maybe there should be a free-standing function `ClassesMap makeClassesMap(ExtractorCallbacks&);` to better show intent? The call-once issue would still persist then, though.